File: py/file_reader/tree_node.py
class TreeNode:
    """
    This objects represent scanned directory in a tree
    """

    def __init__(self, path: str, size: int) -> None:
        self.children: list[TreeNode] = []
        self.path: str = path
        self.__size: int = size

    def append_child(self, child_node) -> None:
        self.children.append(child_node)

File: c_sharp/src/Reader/TreeNode.cs
namespace Reader {
  public class TreeNode {
    public string Path { get; private set; }
    private int Size;
    private List<TreeNode> Children;
    
    public TreeNode(string path, int size = 0)
    {
      this.Path = path;
      this.Size = size;
      this.Children = new List<TreeNode>();
    }

    public void AppendChild(TreeNode node)
    {
      this.Children.Add(node);
    }
  }
}